LienShark proposal for EigenApproWithGersgorin

LienShark opened the pull request:
<!-- Please fill the following checkboxes -->
---
A good PR should include the following parts:

- [ ] Provide a clear title
- [ ] Provide meaningful descriptions in the PR and the commit logs
- [ ] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [ ] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [ ] Exclude irrelevant files, such as binaries, text files, or dot files
- [ ] Make sure tests/CI pass

yungyuc committed:
Add homework assignment 2

lebr0nli committed:
Fix the typo in HW1

yungyuc committed:
Merge pull request #20 from lebr0nli/fix-typo-in-hw1

c1ydehhx committed:
Fix: Check score is 5 or exit code should be non-zero

yungyuc committed:
Merge pull request #27 from ntut-xuan/fix-hw1-ci

Fix: Check score is full mark or exit code should be non-zero

Commit:
term project proposal

yungyuc commented on project/LienShark/Readme.rst:
```diff
@@ -0,0 +1,99 @@
+=================
+Project
+=================
+Efficient-Eigenvalue-Approximation-using-Gersgorin-Theorem-and-PCA-for-Dimensionality-Reduction 
```
The project name needs to be shortened.

yungyuc commented on project/LienShark/Readme.rst:
```diff
@@ -0,0 +1,99 @@
+Basic Information
+=================
+
+GitHub repository :https://github.com/LienShark/Efficient-Eigenvalue-Approximation-using-Gersgorin-Theorem-and-PCA-for-Dimensionality-Reduction
```
Please use a short name for your github repository too.

yungyuc commented on project/LienShark/Readme.rst:
```diff
@@ -0,0 +1,99 @@
+=================
+
+GitHub repository :https://github.com/LienShark/Efficient-Eigenvalue-Approximation-using-Gersgorin-Theorem-and-PCA-for-Dimensionality-Reduction
+About:I will try to use Gersgorin Theorem to find eigenvalue approximation to do spectral,
```
Spectral and principal value analysis are classical numerical methods.  You can tell a lot of stories with them.

yungyuc commented on project/LienShark/Readme.rst:
```diff
@@ -0,0 +1,99 @@
+
+Prospective Users
+=================
+I expect that the calculation time can be shortened and efficiency improved in data-intensive research 
```
What is your baseline of runtime performance?

yungyuc commented on project/LienShark/Readme.rst:
```diff
@@ -0,0 +1,99 @@
+===================
+
+1.input and output
+  input:massive matrix
```
How do you address majoring?

yungyuc commented on project/LienShark/Readme.rst:
```diff
@@ -0,0 +1,99 @@
+  2.How to narrow down the possible range of eigenvalues
+  3.knowledage about NLP and ML
+  4.knowledage about Finance and genetic information
+* Week 1 (10/21): Write preliminary architecture and simple algorithm implementation
```
Revise to include automatic testing and CI early.

yungyuc reviewed (changes requested):
When requesting for a next review, please leave a global comment summarizing your change.

- [x] [L4](https://github.com/yungyuc/nsdhw_24au/pull/47/files#diff-fec19e96fbfc351f95b3ac05432fbca60819a9075c25d6a8e59b4b783929c7aaR4): The project name is too long.  Please use no more than 20 characters.
- [x] [L31](https://github.com/yungyuc/nsdhw_24au/pull/47/files#diff-fec19e96fbfc351f95b3ac05432fbca60819a9075c25d6a8e59b4b783929c7aaR31): Clarify the runtime performance baseline.  Add it to the proposal.
- [ ] [L46](https://github.com/yungyuc/nsdhw_24au/pull/47/files#diff-fec19e96fbfc351f95b3ac05432fbca60819a9075c25d6a8e59b4b783929c7aaR46): Specify how to deal with matrix majoring.
- [x] [L84](https://github.com/yungyuc/nsdhw_24au/pull/47/files#diff-fec19e96fbfc351f95b3ac05432fbca60819a9075c25d6a8e59b4b783929c7aaR84): Plan early for automatic testing and CI.

yungyuc commented:
Please leave an entry in E3.  I am not seeing it.

Commit:
Solve the teacher’s comments just now

yungyuc commented:
Reminder:

> When requesting for a next review, please leave a global comment summarizing your change.

Commit:
Improve some points that were not improved last time, including clearly stating what massive means

LienShark commented:
1. I made my title and github URL shorter
2. I changed my goal slightly and changed it to create svd and pca first, hoping to calculate faster than the existing packages!And take the original goal of approximation as an extended challenge, and also describe more sources of ideas about approximation.
3.  As mentioned in the second point, my current main goal is to speed up svd and pca, so my baseline will be placed on the existing python or R package to compare with them.Of course, I update it to my documents
4. I clearly points out what a massive matrix is.
5. Fine-tuned my planning schedule so that I can start automatic testing and CI earlier
6. I also leave an entry in E3.
I complete the all requested
Could you kindly review what I revised in the README? I’d appreciate your feedback.Thank you for your time.

yungyuc commented on project/LienShark/Readme.rst:
```diff
@@ -0,0 +1,129 @@
+================
+First, I will learn how to do SVD decomposition and spectral decomposition for use in PCA.
+I will also survey the current methods of SVD decomposition and spectral decomposition, compare and improve them.
+second,I hope I still have time to add Gersgorin Theorem's method to compare and see if 
```
It is clear to pick a numerical method as a baseline.  You mentioned to use an existing package in [your comment](https://github.com/yungyuc/nsdhw_24au/pull/47#issuecomment-2365136211), but I am not seeing what they are in the proposal.  If you wrote them in the proposal, please leave an inline comment to highlight for review.

It will be hard for you to keep track of items you did not write in the plan.

yungyuc commented on project/LienShark/Readme.rst:
```diff
@@ -0,0 +1,129 @@
+===================
+
+1.input and output
+  input:(massive) matrix (massive I mean is 1 million row or column , since this is approximately the size of the LLM model )
```
I do not see how you will address majoring in the proposal.  Please update.  If you actually wrote it already but I missed it, please leave an inline comment to highlight for review.

I don't think you can create a dense square matrix of 1 million row and column.  Your definition of massive matrix may need adjustment.

yungyuc reviewed (changes requested):
- [ ] Specify how to deal with matrix majoring.
- [ ] Add an entry in `project/README.rst` pointing to your proposal.

ThreeMonth03 commented:
@LienShark By the way, I check your E3 submission. 
You should submit the link `https://github.com/yungyuc/nsdhw_24au/pull/47` instead of the link `https://github.com/LienShark/EigenApproWithGersgorin`. Could you please modify your submission on E3?

Commit:
Merge remote-tracking branch 'upstream/master' into LienShark-project-attempt1
"I need to complete hw2 so I merge teacher's master to my master"

Commit:
Specify how to deal with matrix majoring and add baseline and explain massive matrix more clearly

LienShark commented:
@yungyuc 
1.I plan to use either Incremental SVD or memory mapping to address memory limitations. 
2.I add some description about my term project to readme but I am sorry since I am not very sure what is teacher's mean about  " an entry in project/README.rst pointing to your proposal." . it is right to add description? or still something I need to adjust?
3.I updated my baseline , and they are numpy.linalg.svd()  and  sklearn.decomposition.PCA()
4.I explain something about my definition of massive matrix in my input

@ThreeMonth03 
Sorry, I will modify it, thank you for your remind 
But  I find that I can not modify it in E3.Can you please adjust the relevant permissions? Thank you TA

yungyuc reviewed (changes requested):
@LienShark , please clean up your commit histories before requesting the next review.  You should rebase to the latest `project` branch.  Your current PR contains changesets coming from other people.

Regarding the entry in `project/README.rst`, please refer to the merged PRs like #54.  If you need help with git, please go to sciwork meetup and/or the TA.

LienShark commented:
Yes, I need some help with my github pull request, but I will be back in Tainan this week (because of the National Day holiday), can I join the meeting again next Wednesday and deal with this matter?
Will it make the problem more complicated if I continue to upload my code before this happens?

ThreeMonth03 commented:
> Yes, I need some help with my github pull request, but I will be back in Tainan this week (because of the National Day holiday), can I join the meeting again next Wednesday and deal with this matter? Will it make the problem more complicated if I continue to upload my code before this happens?

You can join the meeting in every Wednesday night, and you can upload your code at first. It is not complicated to rebase and squash the commit messages.

ThreeMonth03 commented:
> @yungyuc 1.I plan to use either Incremental SVD or memory mapping to address memory limitations. 2.I add some description about my term project to readme but I am sorry since I am not very sure what is teacher's mean about " an entry in project/README.rst pointing to your proposal." . it is right to add description? or still something I need to adjust? 3.I updated my baseline , and they are numpy.linalg.svd() and sklearn.decomposition.PCA() 4.I explain something about my definition of massive matrix in my input
> 
> @ThreeMonth03 Sorry, I will modify it, thank you for your remind But I find that I can not modify it in E3.Can you please adjust the relevant permissions? Thank you TA

I think you could exit your link now. Please modify the link as soon as possible.

LienShark commented:
> > @yungyuc 1.I plan to use either Incremental SVD or memory mapping to address memory limitations. 2.I add some description about my term project to readme but I am sorry since I am not very sure what is teacher's mean about " an entry in project/README.rst pointing to your proposal." . it is right to add description? or still something I need to adjust? 3.I updated my baseline , and they are numpy.linalg.svd() and sklearn.decomposition.PCA() 4.I explain something about my definition of massive matrix in my input
> > @ThreeMonth03 Sorry, I will modify it, thank you for your remind But I find that I can not modify it in E3.Can you please adjust the relevant permissions? Thank you TA
> 
> I think you could exit your link now. Please modify the link as soon as possible.


I just updated my assignment at E3,Thanks TA

yungyuc commented:
> Yes, I need some help with my github pull request, but I will be back in Tainan this week (because of the National Day holiday), can I join the meeting again next Wednesday and deal with this matter? Will it make the problem more complicated if I continue to upload my code before this happens?

The proposal due time was 9/19.  It is 10/7 today.  It is already late.  Please get it fixed as soon as possible.  I may stop accepting proposal any time.

yungyuc commented:
Please correct this PR.  If you have further questions, go to the nsd discord channel to seek help as soon as possible.

yungyuc commented:
I am reopening this PR for you, @LienShark

yungyuc commented:
@Sean20405, I wonder if you may be available to help @LienShark with his Github issue?

ThreeMonth03 commented:
@yungyuc It seems that he pushs the new branch forcely, so he cannot reopen this pull request directly. I'm curious whether he could reopen this pull request.

yungyuc commented:
@LienShark you should not have deleted the branch `LienShark-project-attempt1`.  Now you are in a deadlock.

I don't like you to use a different PR because it will be inconsistent to the grading records elsewhere, but if you have to use a different PR, please work it out as soon as possible to address all points I have raised in the previous review.  Add annotations to how you address the points and request for review.

yungyuc commented:
> @yungyuc It seems that he pushs the new branch forcely, so he cannot reopen this pull request directly. I'm curious whether he could reopen this pull request.

Unfortunately, it seems to be the case.  @LienShark, please do your best get it right as soon as possible.

yungyuc commented:
@LienShark and try if this works: https://stackoverflow.com/questions/59394966/deleted-and-recreated-the-branch-for-a-pull-request-on-github-how-can-i-reopen

LienShark commented:
I'm sorry about deleting the branch LienShark-project-attempt1, because I just had to take a break from some work in the lab that needed to be processed. I will finish it as soon as possible! Thank you so much to @yungyuc and @ThreeMonth03

Sean20405 commented:
> @Sean20405, I wonder if you may be available to help @LienShark with his Github issue?

Sorry for the late reply. Sure! But it seems that the problem now becomes restoring the deleting branch XD.

(I'm not sure whether it would help now but) For the original issue, I've met when submitting my proposal. I used the method you mentioned in DC (rebase and drop the commit), and then force pushing the change by `-f`. Btw you could also use `rebase` to orginize the commit (I found it very useful recently).

As for your current problem, maybe others in DC have answered it completely. (I haven't used it before so I'm not sure whether it will work, but it sounds reasonable XD)

yungyuc commented:
> > @Sean20405, I wonder if you may be available to help @LienShark with his Github issue?
> 
> Sorry for the late reply. Sure! But it seems that the problem now becomes restoring the deleting branch XD.

No problem.  Thanks for helping.  Did @LienShark find you to discuss?

yungyuc commented:
This PR is succeeded by #94 .
